OasisDEX/oasis-borrow
# Auto-take Profit changes information

# Auto-take Profit changes information

All of the calculations are taken from multiply vault, but since they are trapped inside of a pipeline that doesn't allow any access from outside of it, I needed to take out some of the stuff out and create separated methods for it.

There are couple differences in changes information compared to figma designs, all of it were consulted:
- Numbers are displayed in full notations since we aren't using short notation anywhere,
- slippage, ETH price and price impact were removed,
- debt is displayed in DAI, not in $,
- estimated transaction costs are different (see: https://discord.com/channels/837076147694207067/839057646480785438/1024272474562633838).

I've also prepared `getEstimatedCostOnClose` thinking that it is going to be used in Stop-Loss and there is going to be new one, similar, `getEstimatedCostOnAdjustMultiply`.

![image](https://user-images.githubusercontent.com/16230404/194321433-1b3e8e68-d42d-48b8-ad75-51c0932d669f.png)
  
## Changes 👷‍♀️

- Filled estimated profit with data calculated from UI state,
- filled changes infromation width data calculated from UI state,
- connected Auto-take Profit sidebar to missing generic methods for title, cta label etc.,
- created `estimatedCostOnClose` and `estimatedProfitOnClose` as boilerplate methods to be used in the rest of auto features.
  
## How to test 🧪

Try to add Auto-take Profit trigger and see how changes in form are reflected on summary.
